packages/jupyter-ai-magics/jupyter_ai_magics/providers.py

```diff
@@ -728,6 +730,84 @@ async def _acall(self, *args, **kwargs) -> Coroutine[Any, Any, str]:
         return await self._call_in_executor(*args, **kwargs)
 
 
+class OllamaProvider(BaseProvider, Ollama):
+    id = "ollama"
+    name = "Ollama"
+    model_id_key = "model"
+    # List of available models
+    models = [
+        "gemma",
+        "gemma2",
+        "llama2",
+        "llama3",
+        "llama3:70b",
+        "phi3",
+        "mistral",
+        "tinyllama",
+        "qwen2",
+        "qwen2:7b",
+        "qwen2:72b",
+    ]
+    # Default base URL for the API
+    base_url = "http://localhost:11434"
+    # Fields for configuring the provider
+    fields = [
+        TextField(
+            key="ollama_api_base", label="Base API URL (optional)", format="text"
+        ),
+    ]
+
+    def __init__(self, **kwargs):
+        # Extract 'ollama_api_base' from kwargs, if not present, it returns None
+        ollama_api_base = kwargs.pop("ollama_api_base", None)
+        # Initialize the parent class with the remaining kwargs
+        super().__init__(**kwargs)
+        # Set the base_url to ollama_api_base if provided, otherwise use the default base_url
+        self.base_url = ollama_api_base or self.base_url
+        # Create a session for making requests
+        self.session = requests.Session()
+
+    def _send_request(self, endpoint: str, data: dict) -> dict:
+        """Send a POST request to the specified Ollama API endpoint."""
+        url = f"{self.base_url}/{endpoint}"
+        print("url is : ", url)
```

Comment:
Do you need this print statement? If so, use `self.log.info()` as is the approach in much of the code base.